acceptance-tests/src/test/java/it/com/atlassian/bitbucket/jenkins/internal/SmokeTest.java

```diff
@@ -0,0 +1,444 @@
+package it.com.atlassian.bitbucket.jenkins.internal;
+
+import io.restassured.RestAssured;
+import io.restassured.path.json.JsonPath;
+import io.restassured.response.Response;
+import io.restassured.specification.RequestSpecification;
+import it.com.atlassian.bitbucket.jenkins.internal.pageobjects.*;
+import it.com.atlassian.bitbucket.jenkins.internal.util.BitbucketUtils.*;
+import org.apache.commons.io.FileUtils;
+import org.apache.commons.io.IOUtils;
+import org.eclipse.jgit.api.Git;
+import org.eclipse.jgit.api.errors.GitAPIException;
+import org.eclipse.jgit.lib.TextProgressMonitor;
+import org.eclipse.jgit.revwalk.RevCommit;
+import org.eclipse.jgit.transport.CredentialsProvider;
+import org.eclipse.jgit.transport.UsernamePasswordCredentialsProvider;
+import org.hamcrest.Description;
+import org.hamcrest.TypeSafeDiagnosingMatcher;
+import org.jenkinsci.test.acceptance.junit.AbstractJUnitTest;
+import org.jenkinsci.test.acceptance.junit.WithPlugins;
+import org.jenkinsci.test.acceptance.plugins.credentials.CredentialsPage;
+import org.jenkinsci.test.acceptance.plugins.credentials.UserPwdCredential;
+import org.jenkinsci.test.acceptance.po.Build;
+import org.jenkinsci.test.acceptance.po.FreeStyleJob;
+import org.jenkinsci.test.acceptance.po.JenkinsConfig;
+import org.jenkinsci.test.acceptance.po.WorkflowJob;
+import org.junit.Before;
+import org.junit.Ignore;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.TemporaryFolder;
+
+import java.io.File;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.PrintWriter;
+import java.net.URLEncoder;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+
+import static io.restassured.http.ContentType.JSON;
+import static it.com.atlassian.bitbucket.jenkins.internal.util.BitbucketUtils.*;
+import static java.nio.charset.StandardCharsets.UTF_8;
+import static java.time.Duration.ofMinutes;
+import static java.util.UUID.randomUUID;
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.is;
+import static org.jenkinsci.test.acceptance.plugins.credentials.ManagedCredentials.DEFAULT_DOMAIN;
+import static org.jenkinsci.test.acceptance.po.Build.Result.SUCCESS;
+
+@WithPlugins({"mailer", "atlassian-bitbucket-server-integration"})
+public class SmokeTest extends AbstractJUnitTest {
```

Comment:
We can make use of few infra classes that were created as part of build status posting refactoring. For example, 

1. `it.com.atlassian.bitbucket.jenkins.internal.fixture.GitHelper` - Will help in removing boiler plate around creating Jenkins file and checking in. 
2. `it.com.atlassian.bitbucket.jenkins.internal.fixture.JenkinsProjectHandler` - Will help in removing boiler plate code around different type of jobs, force scanning for changes, force scanning for branches. 

This will make this test more readable.

Reply:
Looking more into it, I can see that you are using acceptance test framework method to do `it.com.atlassian.bitbucket.jenkins.internal.fixture.JenkinsProjectHandler` work. However, `GitHelper` can still be used

Reply:
As discussed in chat and [in this thread](https://github.com/jenkinsci/atlassian-bitbucket-server-integration-plugin/pull/179#discussion_r455029864), the two modules will stay completely separate for now, which makes reusing things like `GitHelper` impossible until we address the bigger issue of a common (sub-)module/shared dependencies.